dubbo-metrics/dubbo-metrics-default/src/main/java/org/apache/dubbo/metrics/collector/sample/ThreadPoolMetricsSampler.java

```diff
@@ -64,6 +69,20 @@ private void collect() {
             }
         }catch(Exception ex){}
 
+        if (this.dataStore == null) {
+            this.dataStore = collector.getApplicationModel().getExtensionLoader(DataStore.class).getDefaultExtension();
```

Comment:
please add NullPointException check

9Z [ERROR] org.apache.dubbo.metrics.filter.MetricsFilterTest.testMissingGroupAndVersion  Time elapsed: 0.041 s  <<< ERROR!
2023-02-18T01:17:26.2131813Z java.lang.NullPointerException: Cannot invoke "org.apache.dubbo.rpc.model.ApplicationModel.getExtensionLoader(java.lang.Class)" because the return value of "org.apache.dubbo.metrics.collector.DefaultMetricsCollector.getApplicationModel()" is null
2023-02-18T01:17:26.2132911Z 	at org.apache.dubbo.metrics.collector.sample.ThreadPoolMetricsSampler.collect(ThreadPoolMetricsSampler.java:73)
2023-02-18T01:17:26.2133674Z 	at org.apache.dubbo.metrics.collector.sample.ThreadPoolMetricsSampler.sample(ThreadPoolMetricsSampler.java:53)
2023-02-18T01:17:26.2134391Z 	at org.apache.dubbo.metrics.collector.DefaultMetricsCollector.collect(DefaultMetricsCollector.java:97)
2023-02-18T01:17:26.2135048Z 	at org.apache.dubbo.metrics.filter.MetricsFilterTest.getMetricsMap(MetricsFilterTest.java:327)
2023-02-18T01:17:26.2135741Z 	at org.apache.dubbo.metrics.filter.MetricsFilterTest.testMissingGroupAndVersion(MetricsFilterTest.java:284)
2023-02-18T01:17:26.2136378Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
2023-02-18T01:17:26.2136970Z 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)